openassessment/xblock/staff_info_mixin.py

```diff
@@ -204,9 +214,13 @@ def render_student_info(self, data, suffix=''): # pylint: disable=W0613
         Must be course staff to render this view.
 
         """
-        student_id = data.params.get('student_id', '')
-        path, context = self.get_student_info_path_and_context(student_id)
-        return self.render_assessment(path, context)
+        try:
+            student_id = data.params.get('student_id', '')
+            path, context = self.get_student_info_path_and_context(student_id)
+            return self.render_assessment(path, context)
+
+        except PeerAssessmentInternalError as ex:
+            return self.render_error(self._(ex.message))
```

Comment:
Unfortunately, I don't think this will work. ugettext will parse the code pre-compiled to find the strings within the function, so you'd need the string in-line here, rather than ex.message.

If we're using this pattern elsewhere, we should probably fix it. (I may have done this at times before I knew the caveats of ugettext)